docs/src/pages/components/data-grid/demo/FullFeauredDemo.tsx

```diff
@@ -0,0 +1,304 @@
+      backgroundColor: '#1890ff',
+      borderColor: '#1890ff',
+    },
+    '& .MuiCheckbox-root.Mui-checked .MuiIconButton-label:after': {
+      position: 'absolute',
+      display: 'table',
+      border: '2px solid #fff',
+      borderTop: 0,
+      borderLeft: 0,
+      transform: 'rotate(45deg) translate(-50%,-50%)',
+      opacity: 1,
+      transition: 'all .2s cubic-bezier(.12,.4,.29,1.46) .1s',
+      content: '""',
+      top: '50%',
+      left: '39%',
+      width: 5.71428571,
+      height: 9.14285714,
+    },
+    '& .MuiCheckbox-root.MuiCheckbox-indeterminate .MuiIconButton-label:after': {
+      width: 8,
+      height: 8,
+      backgroundColor: '#1890ff',
+      transform: 'none',
+      top: '39%',
+      border: 0,
+    },
+  },
+}));
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    display: 'flex',
+    flexDirection: 'column',
+    height: 600,
+    width: '100%',
+    '& .MuiFormGroup-options': {
+      alignItems: 'center',
+      paddingBottom: theme.spacing(1),
+      '& fieldset': {
+        minWidth: 100,
+        margin: theme.spacing(2, 2, 2, 0),
+      },
+    },
+  },
+}));
+
+type GridDataType = 'Employee' | 'Commodity';
+type GridDataThemeOption = 'default' | 'ant';
+
+interface GridPaginationSettings {
+  pagination: boolean;
+  autoPageSize: boolean;
+  pageSize: number | undefined;
+}
+
+interface GridConfigOptions {
+  size: number;
+  type: GridDataType;
+  pagesize: number;
+  theme: GridDataThemeOption;
+}
+
+interface GridToolbarContainerProps {
+  onApply: (options: GridConfigOptions) => void;
+  size: number;
+  type: GridDataType;
+  theme: GridDataThemeOption;
+}
+
+function SettingsPanel(props: GridToolbarContainerProps) {
+  const { onApply, type, size, theme } = props;
+  const [sizeState, setSize] = React.useState<number>(size);
+  const [typeState, setType] = React.useState<GridDataType>(type);
+  const [
+    selectedPaginationValue,
+    setSelectedPaginationValue,
+  ] = React.useState<number>(-1);
+  const [activeTheme, setActiveTheme] = React.useState<GridDataThemeOption>(theme);
+
+  const handleSizeChange = React.useCallback((event) => {
+    setSize(Number(event.target.value));
+  }, []);
+
+  const handleDatasetChange = React.useCallback((event) => {
+    setType(event.target.value);
+  }, []);
+
+  const handlePaginationChange = React.useCallback((event) => {
+    setSelectedPaginationValue(event.target.value);
+  }, []);
+
+  const handleThemeChange = React.useCallback((event) => {
+    setActiveTheme(event.target.value);
+  }, []);
+
+  const handleApplyChanges = React.useCallback(() => {
+    onApply({
+      size: sizeState,
+      type: typeState,
+      pagesize: selectedPaginationValue,
+      theme: activeTheme,
+    });
+  }, [sizeState, typeState, selectedPaginationValue, activeTheme, onApply]);
+
+  return (
+    <FormGroup className="MuiFormGroup-options" row>
+      <FormControl component="fieldset">
```

Comment:
```suggestion
      <FormControl>
```

x the other, this is not a fieldset, it would need a legend to be correct

Reply:
removed